### PR TITLE
announce the line to screen readers when `Go to line` is used

### LINKED ISSUE #179283: [Accessibility]: Make "Go to line" announce focused line after executed
CC @meganrogge

This is not a regression.

If you use "Go to line (`Ctrl+G`)" and land
in a specific line, screen reader does not read the current line b/c there's no announcement.

You may recall this similar issue that you encountered when addressing Alt+F5 (go to the next diff) did not announce the focused line and you used a selection to fire an announcement for screen readers.

I have found that `editor.action.wordHighlight.next` (F7) auto-announce the next focus when this key is pressed. This command may be worth looking into. I am not sure whether it also used the selection though.

### PR DESCRIPTION
**Pull request by meganrogge**
fixes https://github.com/microsoft/vscode/issues/179283

### CODE REVIEW

**Review by Tyriar — APPROVED** [review]

File: src/vs/editor/contrib/quickAccess/browser/editorNavigationQuickAccess.ts [hunk 1]
```diff
@@ -146,6 +147,10 @@ export abstract class AbstractEditorNavigationQuickAccessProvider implements IQu
 		if (!options.preserveFocus) {
 			editor.focus();
 		}
+		const model = editor.getModel();
+		if (model && 'getLineContent' in model) {
```
Comment (bpasero) [inline src/vs/editor/contrib/quickAccess/browser/editorNavigationQuickAccess.ts:151]: @meganrogge why not use the existing method `getModel` here that does the cast for you?

https://github.com/microsoft/vscode/blob/ada796969b6599a66c6deb0d057aae4b4f3ecf4b/src/vs/editor/contrib/quickAccess/browser/editorNavigationQuickAccess.ts#L156-L160

**Comment by isidorn** [comment]
@bpasero thanks for a great review and great questions asked.
@meganrogge do we really need `alert` here? Why is screen reader not reading content when Go To Line navigates to a new line? This should work. If needed, we can also reach out to Screen Reader folk to understand this better.

**Comment by meganrogge** [comment]
Yes, thanks for the questions and input. This didn't work on either windows or Mac before these changes. I used `alert` because typically that's what we do when the screen reader is not behaving as expected/ reading the content correctly. to me, it looks like the virtual cursor (black outline) has a selection size of 0, which could explain it. Maybe @alexdima would know more

https://user-images.githubusercontent.com/29464607/232565703-33fb156b-16ed-40bf-8288-158b931fa9b2.mov

**Comment by bpasero** [comment]
But it seems really wrong to me to step in and do something that the editor knows best: how to announce the current cursor position. I would have not expected a call to `alert` from the picker but a method from the editor to do exactly the same that would happen when you change selection and get that announced.

**Comment by meganrogge** [comment]
Ok, I'll try to figure out why selection does not occur when it is initiated from the command palette whereas it works when using arrow keys. I had figured that was by design/ not something we want to change

**Comment by bpasero** [comment]
Maybe interview @hediet or @alexdima too, maybe programmatic selection changes are treated differently from keyboard interactions. I remember that the selection change events have a `source` property that can be `api` or `keyboard` and maybe we reason differently based on that.

**Comment by meganrogge** [comment]
https://user-images.githubusercontent.com/29464607/232568191-7276fbac-929a-4278-a5c8-b2ac1acf46b6.mov

**Comment by bpasero** [comment]
Why would that not trigger the alert? An empty selection is still a line change, no?

**Comment by meganrogge** [comment]
like I was saying, that's why for the diff editor, go to next / previous difference selects the line so the screen reader will read it

**Comment by meganrogge** [comment]
@hediet & @alexdima would appreciate your thoughts here. IMO, we don't want to alter the selection just so the screen reader gets the content bc that is pretty hacky

**Comment by bpasero** [comment]
Let us continue the discussion in https://github.com/microsoft/vscode/issues/180144

**Pull request merged**